### PR TITLE
Add documentation for PEP 735 support

### PR DESCRIPTION
**Pull request by zanieb**
This adds the minimal documentation I think we need to release PEP 735 support.

I want to add documentation on `include-group` and such but that can come after.

I also want to restructure some of the project dependency documentation, but that will be easier once this all lands in `main`.

### CODE REVIEW

File: docs/concepts/dependencies.md [hunk 1]
```diff
@@ -393,23 +393,91 @@ $ uv add httpx --optional network
...
+
+Before `[dependency-groups]` was standardized, uv used the `tool.uv.dev-dependencies` field to
+specify development dependencies, e.g.:
+
+```toml title="pyproject.toml"
+[tool.uv]
+dev-dependencies = [
+  "pytest"
+]
+```
```
Comment (MichaReiser) [inline docs/concepts/dependencies.md:476]: What happens if I run `uv add --dev xx` when there's only a `dev-dependency` but no `dev` dependency group?

Comment (zanieb) [reply docs/concepts/dependencies.md:476]: That's a great question, Charlie is fixing the behavior there (right now we actually never add to the `dev` group during `uv add --dev`) so I'm not sure what the final behavior is yet. I'll update this to include that once he puts up that PR.

File: docs/concepts/dependencies.md [hunk 2]
```diff
@@ -393,23 +393,91 @@ $ uv add httpx --optional network
...
+```toml title="pyproject.toml"
+[dependency-groups]
+dev = [
+  "pytest"
+]
+lint = [
+  "ruff"
+]
+```
+
```
[the remark this answers is not in the document]
Comment (zanieb) [reply docs/concepts/dependencies.md:440]: Will do!

#8567 also adds some context in the CLI about the equivalence of these commands.

**Review by charliermarsh — APPROVED** [review]

File: docs/concepts/dependencies.md (hunk 1 above)
Comment (charliermarsh) [reply docs/concepts/dependencies.md:476]: (It adds to `dev-dependencies`.)

**Pull request merged**